indico/modules/events/abstracts/controllers/abstract_list.py

```diff
@@ -107,11 +110,34 @@ def _process(self):
 
 
 class RHCreateAbstract(RHAbstractListBase):
-    def _process(self):
+    """Create an abstract from scratch or with existing cloned data"""
+
+    @staticmethod
+    def clone_fields(abstract):
+        field_names = ['title', 'description', 'submission_comment', 'submitted_for_tracks', 'submitted_contrib_type']
+        field_data = {}
+        for f in field_names:
+            field_data[f] = getattr(abstract, f)
+        person_links = []
+        link_attrs = get_simple_column_attrs(AbstractPersonLink)
+        for old_link in abstract.person_links:
+            link = AbstractPersonLink(person=old_link.person)
+            link.populate_from_attrs(old_link, link_attrs)
+            person_links.append(link)
+        field_data['person_links'] = person_links
+        for f in abstract.field_values:
+            field_data[f'custom_{f.contribution_field_id}'] = f.data
+        return field_data
+
+    @use_kwargs({'abstract': ModelField(Abstract, data_key='abstract_id')}, location='query')
```

Comment:
:warning: This lets you clone abstracts from other events as well!

Reply:
Nice catch!

I noticed there is a `get_query` for it, do you think it would be more reasonable to go back to the `_process_args` method or have a new decorator (with `get_query=.with_parent(...)`) here given the abstract isn't always there?

Reply:
It won't really work with the decorator since you don't have access to the event there - so I think using the webargs/marshmallow field here isn't that useful; you'd need to manually parse it with webargs within the method (like we sometimes to in the editing module) :/